Rubberduck.Core/UI/CodeExplorer/Commands/Abstract/CodeExplorerMoveToFolderCommandBase.cs

```diff
@@ -49,10 +49,8 @@ public abstract class CodeExplorerMoveToFolderCommandBase : CodeExplorerCommandB
             AddToCanExecuteEvaluation(SpecialEvaluateCanExecute);
         }
 
-        private bool SpecialEvaluateCanExecute(object parameter)
-        {
-            return _parserStatusProvider.Status == ParserState.Ready;
-        }
+        protected override bool SpecialEvaluateCanExecute(object parameter) => 
+            _parserStatusProvider.Status == ParserState.Ready;
```

Comment:
This feels wrong. Look at the code here:
https://github.com/rubberduck-vba/Rubberduck/blob/51f2d139faf7c245b96e09bcd31b4424491dc186/Rubberduck.Core/UI/CodeExplorer/Commands/CodeExplorerCommandBase.cs#L11-L15

The design is to allow the derived classes to _add_ conditions instead of overriding the conditions. That enables us to enforce a condition in effect by parent without having the derived child overriding it and thus ignoring the condition. That is especially important because this derives from `ComCommand` which has this condition:
https://github.com/rubberduck-vba/Rubberduck/blob/51f2d139faf7c245b96e09bcd31b4424491dc186/Rubberduck.Core/UI/Command/ComCommands/ComCommandBase.cs#L9-L18
to ensure that all commands that interacts with VBIDE will no longer operate while VBIDE is shutting down and thus prevent AV.

Reply:
I will revert this and rename it to `CanExecuteInternal` in the derived commands, like here:

        public PeekDefinitionCommand(CodeExplorerDockablePresenter codeExplorer, IVbeEvents vbeEvents, ISelectedDeclarationProvider selection)
            : base(vbeEvents)
        {
            PopupProvider = (codeExplorer.UserControl as CodeExplorerWindow)?.ViewModel;
            SelectedDeclarationProvider = selection;
            AddToCanExecuteEvaluation(CanExecuteInternal);
        }

Actually `EvaluateCanExecute` is probably a better name.

Usually a same-name method hitting a breakpoint in the base class but not in the derived class when debugging, points to unintended member shadowing going on: it tripped me here.

Reply:
Hmm. Both should have been hit, though so not hitting the breakpoint on the derived is also wrong. Just to confirm, did it not get hit after you F5 through the base's breakpoint? The way it's setup, the parent gets its condition evaluated first before the child's and the child's won't be evaluated if the parent's condition fails but _must_ be evaluated if the parent's condition passes.